### ISSUE TITLE
Embeded nodejs in Win32 application

### ISSUE THREAD

**Issue by 823639792** [nodejs/help#585]
@refack

node version: 7.8.0
platform: windows7

I want to embed nodejs in my application.
but when I use  http interface throw exception:
```js
function getServerFileContent(url, sb, fb){
	var http = global.require('http');
	var buffer = new Buffer("");
	var list = [];
	var request = http.get(url, function (res) {
		res.on('data', function (data) {
			list.push(data);
		}).on('end', function (data) {
			buffer = Buffer.concat(list);
			sb(url, buffer);
		}).on("error", function(data){
			fb(url);
		});
	});
	request.on("error", function(){
		fb(url);
	})
}

getServerFileContent("http://www.xxxxx.com/base.js", function(url, data){console.log(url, data.toString())});
```
```
_http_outgoing.js:721 Uncaught TypeError: socket.cork is not a function
    at ClientRequest._flushOutput (_http_outgoing.js:721)
    at ClientRequest._flush (_http_outgoing.js:700)
    at _http_client.js:249
    at callSocketMethod (_http_client.js:656)
    at ClientRequest.onSocket (_http_client.js:661)
    at Object.onceWrapper (events.js:293)
    at emitOne (events.js:96)
    at ClientRequest.emit (events.js:191)
    at tickOnSocket (_http_client.js:622)
    at onSocketNT (_http_client.js:638)
```

**Comment by 823639792** [comment]
in stdio.js  I create dummy stream.
```
var dummyOutput = true;
var W = require('_stream_writable');
var R = require('_stream_readable');
var util = require('util');
util.inherits(DummyWritableStream, W);
util.inherits(DummyReadableStream, R);

var fs = require("fs");
var log = fs.createWriteStream('c:\\log.txt', { 'flags': 'a' });

function DummyWritableStream() {
    W.apply(this, arguments);
    this.buffer = [];
    this.written = 0;
}

function DummyReadableStream() {
    R.apply(this, arguments);
    this.buffer = [];
    this.reading = false;
}

DummyWritableStream.prototype._write = function (chunk, encoding, cb) {
    //this.buffer.push(chunk.toString());
    //this.written += chunk.length;
    log.write(chunk);
    cb();
};
DummyReadableStream.prototype._read = function (param) {
    log.write("_read");
}

function createWritableDummyStream(fd) {
    var stream = new DummyWritableStream();
    stream.fd = fd;
    stream._isStdio = true;
    stream.isTTY = false;
    return stream;
}

function createReadableDummyStream(fd) {
    var stream = new DummyReadableStream();
    stream.fd = fd;
    stream._isStdio = true;
    stream.isTTY = false;
    stream.readable = false;
    return stream;
}
```

**Comment by refack** [comment]
@823639792 Sorry thing dropped off my radar... Back in focus.
Did you make any progress?

**Comment by kanbang** [comment]
how to embed nodejs in win32 non-console application (for example：mfc)?
could someone kind show me a tiny demo

**Comment by gireeshpunathil** [comment]
Many design elements of node.js assume that it is a console application. Examples: REPL, standard streams, parent-child communication channels, and the `Console` object itself. I am not sure it can be embedded into an MFC application without making fair amount of refactoring.

**Comment by 823639792** [comment]
I embed nodejs in my mfc application, but inspect by chrome or some other conditions, it will crash.
I modified many source code:
1、stream(stdio.js)
2、inspector
3、Console
4、Run a timeInterval to keep thread;
I suggest you embed V8, or Reference nw.js.
@kanbang

**Comment by refack** [comment]
https://github.com/nodejs/node/issues/14158 - to try to raise focus on the embedding story

**Comment by kanbang** [comment]
@823639792 
I had embed v8 in a draw toolbox, since already using V8, why not embed node.js instead? This would allow developers to use any of the 100,000+ packages in npm without restriction.
![image](https://user-images.githubusercontent.com/4978439/28497671-6492daa2-6fbf-11e7-9fea-8394327fdb3b.png)

could you help me to build a tiny mfc demo, just a helloworld, win32 non-console, without child process

**Comment by styfle** [comment]
@refack @Trott Should this ticket be closed in favor of https://github.com/nodejs/node/issues/14158 ?

**Comment by Trott** [comment]
> @refack @Trott Should this ticket be closed in favor of nodejs/node#14158 ?

Closing. If anyone thinks it should remain open, leave a comment (or re-open it if GitHub let's you). Thanks!
